### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by smusali**
Changes to `README.md`:
- New Environment Variables;
- Change Log.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
 * __LOGDNA_URL__: Specific Endpoint to Stream Log into, *Optional*
+  __Default__: `logs.logdna.com/logs/ingest`
+* __VERBOSE__: Enabling or Disabling to Log `LogSpout` Container, *Optional*
+  __Default__: Enabled
+  __Note__: `0` to Disable
+
+__Note__: Logging `LogSpout` Container is recommended to keep track of the HTTP Request Error or Exceptions.
+
+#### Related to HTTP Client
+* __DIAL_KEEP_ALIVE__: The interval between keep-alive probes for an active network connection.
```
Comment (dchai76) [inline README.md]: Suggest you add "in seconds" somewhere, same as you do below

File: README.md [hunk 2]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+* __DIAL_TIMEOUT__: The maximum amount of time (in `seconds`) a dial will wait for a connect to complete.
+  __Default__: 60
+  __Source__: [net/dial.go#Timeout](https://github.com/golang/go/blob/master/src/net/dial.go#L27-L39)
+* __IDLE_CONN_TIMEOUT__: The maximum amount of time (in `seconds`) an idle (keep-alive) connection will remain idle before closing itself.
+  __Default__: 60
+  __Source__: [net/http/transport.go#IdleConnTimeout](https://github.com/golang/go/blob/master/src/net/http/transport.go#L213-L217)
+* __HTTP_CLIENT_TIMEOUT__: Time limit (in `seconds`) for requests made by this HTTP Client, *Optional*
+  __Default__: 60
+  __Source__: [net/http/client.go#Timeout](https://github.com/golang/go/blob/master/src/net/http/client.go#L89-L104)
+* __TLS_HANDSHAKE_TIMEOUT__: The maximum amount of time (in `seconds`) waiting to wait for a TLS handshake, *Optional*
```
Comment (dchai76) [inline README.md]: The maximum amount of time (in `seconds`) allowed to wait for a TLS handshake, *Optional*

File: README.md [hunk 3]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+  __Default__: 30
+  __Source__: [net/http/transport.go#TLSHandshakeTimeout](https://github.com/golang/go/blob/master/src/net/http/transport.go#L171-L173)
+
+#### Limits
+* __FLUSH_INTERVAL__: The frequency (in `milliseconds`) to ship the batch of logs, *Optional*
```
Comment (dchai76) [inline README.md]: How frequently batches of logs are sent (in `milliseconds`), *Optional*

File: README.md [hunk 4]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+  __Default__: 250
+* __INACTIVITY_TIMEOUT__: The timeout to detect failure in `Docker API` and restart, *Optional*
```
Comment (dchai76) [inline README.md]: This needs to be worded differently but I'm not sure how - as it is I don't know what it means. Also an explanation of the units.

File: README.md [hunk 5]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+  __Default__: 1m
+  __Note__: More information about the possible values can be found [here](https://github.com/gliderlabs/logspout#detecting-timeouts-in-docker-log-streams).
+* __MAX_BUFFER_SIZE__: The maximum byte size (in `mb`) of the batch to ship to `LogDNA`, *Optional*
+  __Default__: 2
+* __MAX_LINE_LENGTH__: The maximum character length for each line, *Optional*
+  __Default__: 16000
+* __MAX_REQUEST_RETRY__: The maximum # of retries in case of network issue, *Optional*
```
Comment (dchai76) [inline README.md]: The maximum number of retries for requests, when there are request failures, *Optional*

File: README.md (hunk 4 above)
Comment (smusali) [reply README.md]: Added more explanation.

File: README.md (hunk 5 above)
Comment (smusali) [reply README.md]: Retry is per line and mechanism is not to retry the request but add the line back to queue.

**Comment by smusali** [comment]
@dchai76, updated! Can you do second round? Thanks in advance!

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,21 +5,21 @@ This file documents all notable changes in `LogDNA LogSpout Image`. The release
 ## v1.2.0 - Released on May 29, 2019
 
 * Added Tagged Build;
-* Added [semantic versioning](http://semver.org);
-* Added [`CHANGELOG.md`](https://github.com/logdna/logspout/blob/master/CHANGELOG.md);
-* Updated [`LICENSE`](https://github.com/logdna/logspout/blob/master/LICENSE);
+* Added [Semantic Versioning](http://semver.org);
+* Added [CHANGELOG.md](https://github.com/logdna/logspout/blob/master/CHANGELOG.md);
```
Comment (dchai76) [inline CHANGELOG.md]: Actually, this and `LICENSE` below I'm fine leaving the backticks since they're file names

File: README.md [hunk 7]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
 * __HOSTNAME__: Alternative Hostname, *Optional*
-* __TAGS__: Comma-separated List of Tags, *Optional*
+* __TAGS__: Comma-Separated List of Tags, *Optional*
 * __LOGDNA_URL__: Specific Endpoint to Stream Log into, *Optional*
+  * __Default__: `logs.logdna.com/logs/ingest`
+* __VERBOSE__: Enabling or Disabling to Log `LogSpout` Container, *Optional*
+  * __Default__: Enabled
+  * __Note__: `0` to Disable
+
+__Note__: Logging the `LogSpout` Container is recommended to keep track of HTTP Request Error or Exceptions.
```
Comment (dchai76) [inline README.md]: "Errors" (plural)

File: README.md [hunk 8]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+  * __Default__: 60
+  * __Source__: [net/http/client.go#Timeout](https://github.com/golang/go/blob/master/src/net/http/client.go#L89-L104)
+* __TLS_HANDSHAKE_TIMEOUT__: The maximum amount of time (in `seconds`) allowed to wait for a TLS handshake, *Optional*
+  * __Default__: 30
+  * __Source__: [net/http/transport.go#TLSHandshakeTimeout](https://github.com/golang/go/blob/master/src/net/http/transport.go#L171-L173)
+
+#### Limits
+* __FLUSH_INTERVAL__: How frequently batches of logs are sent (in `milliseconds`), *Optional*
+  * __Default__: 250
+* __INACTIVITY_TIMEOUT__: The timeout to detect failure in `Docker API` and restart, *Optional*
```
Comment (dchai76) [inline README.md]: How long to wait for inactivity before declaring failure in the `Docker API` and restarting,

File: README.md [hunk 9]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+  * __Default__: 1m
+  * __Note__: More information about the possible values can be found [here](https://github.com/gliderlabs/logspout#detecting-timeouts-in-docker-log-streams). Also see [`time.ParseDuration`](https://golang.org/pkg/time/#ParseDuration) for valid format as recommended [here](https://github.com/gliderlabs/logspout/blob/e671009d9df10e8139f6a4bea8adc9c7878ff4e9/router/pump.go#L112-L116).
+* __MAX_BUFFER_SIZE__: The maximum byte size (in `mb`) of the batch to ship to `LogDNA`, *Optional*
```
Comment (dchai76) [inline README.md]: Let's get rid of "byte" - to say "byte size" and MB in the same sentence feels off

File: README.md (hunk 8 above)
Comment (smusali) [reply README.md]: I liked this!

**Comment by smusali** [comment]
@dchai76, done! Final round?

File: README.md [hunk 10]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+* __FLUSH_INTERVAL__: How frequently batches of logs are sent (in `milliseconds`), *Optional*
+  * __Default__: 250
+* __INACTIVITY_TIMEOUT__: How long to wait for inactivity before declaring failure in the `Docker API` and restarting, *Optional*
+  * __Default__: 1m
+  * __Note__: More information about the possible values can be found [here](https://github.com/gliderlabs/logspout#detecting-timeouts-in-docker-log-streams). Also see [`time.ParseDuration`](https://golang.org/pkg/time/#ParseDuration) for valid format as recommended [here](https://github.com/gliderlabs/logspout/blob/e671009d9df10e8139f6a4bea8adc9c7878ff4e9/router/pump.go#L112-L116).
+* __MAX_BUFFER_SIZE__: The maximum size (in `mb`) of the batch to ship to `LogDNA`, *Optional*
+  * __Default__: 2
+* __MAX_LINE_LENGTH__: The maximum character length for each line, *Optional*
+  * __Default__: 16000
+* __MAX_REQUEST_RETRY__: The maximum number of retries for sending a line when there are network failure, *Optional*
```
Comment (dchai76) [inline README.md]: "failures" plural

File: README.md [hunk 11]
```diff
@@ -6,40 +6,85 @@ A Docker LogSpout image to stream logs from your containers to LogDNA.
...
+  * __Default__: 30
+  * __Source__: [net/http/transport.go#TLSHandshakeTimeout](https://github.com/golang/go/blob/master/src/net/http/transport.go#L171-L173)
+
+#### Limits
+* __FLUSH_INTERVAL__: How frequently batches of logs are sent (in `milliseconds`), *Optional*
+  * __Default__: 250
+* __INACTIVITY_TIMEOUT__: How long to wait for inactivity before declaring failure in the `Docker API` and restarting, *Optional*
+  * __Default__: 1m
+  * __Note__: More information about the possible values can be found [here](https://github.com/gliderlabs/logspout#detecting-timeouts-in-docker-log-streams). Also see [`time.ParseDuration`](https://golang.org/pkg/time/#ParseDuration) for valid format as recommended [here](https://github.com/gliderlabs/logspout/blob/e671009d9df10e8139f6a4bea8adc9c7878ff4e9/router/pump.go#L112-L116).
+* __MAX_BUFFER_SIZE__: The maximum size (in `mb`) of the batch to ship to `LogDNA`, *Optional*
```
Comment (dchai76) [inline README.md]: The maximum size (in `mb`) of batches to ship to `LogDNA`

**Comment by smusali** [comment]
Updated, @dchai76!

**Comment by gregswift** [comment]
fwiw +1

that being said... we really should have squashed these commits.

**Pull request merged**